app/javascript/utils/form/final_form/MultiDateField.js

```diff
@@ -40,53 +40,65 @@ export const MultiDateField = ({
   })
   let errorField
   if (formName) {
-    errorField = form.getState().errors[formName] ? form.getState().errors[formName][index] : null
+    if (form.getState().errors[formName]) {
+      errorField =
+        index !== null ? form.getState().errors[formName][index] : form.getState().errors[formName]
+    } else {
+      errorField = null
+    }
   } else {
     errorField = form.getState().errors[fieldName.split('.')[0]]
   }
   const baseFieldName = last(fieldName.split('.'))
   const error =
-    form && touched && errorField && errorField[baseFieldName] && errorField[baseFieldName].all
+    form &&
+    touched &&
+    errorField &&
+    ((errorField[baseFieldName] && errorField[baseFieldName].all) || errorField.all)
```

Comment:
nit: can be `errorField[baseFieldName]?.all ||`
```suggestion
    (errorField[baseFieldName]?.all || errorField.all)
```